script.js

```diff
@@ -1,363 +1,12 @@
-          document.getElementById('show-name').innerHTML = `${element.name}`;
-          searchForEpisodes.style.display = 'block';
-      }
-
-    })
-
-  });
- 
-}
-
-
-// An event listener wrapped inside a function to dynamically update the web page while a user is typing in the search bar.
-function showSearch () { 
-
-  const searchInfo = document.querySelector('.search-info');
-  const searchInfo2 = document.querySelector('.search-info2');
-  
-  searchBar.addEventListener('keyup', (e) => {
-
-    // Revealing the hidden information lines
-    searchInfo.style.display = "block";
-    searchInfo2.style.display = "block";
-    
-    //Filtering search results
-    let searchValue = e.target.value.toLowerCase();
-   
-    let searchResult = data.filter((element) => {
-      return (element.name.toLowerCase().includes(searchValue) || element.summary.toLowerCase().includes(searchValue) || element.genres.toString().toLowerCase().includes(searchValue));
-    })  
-    while(rootShows.firstChild) {
-      rootShows.removeChild(rootShows.firstChild)
-    }
-    allShows(searchResult)
-  
-    if (searchValue === "") {
-      searchInfo.style.display = "none";
-      searchInfo2.style.display = "none";
-      window.location.reload();
-    }
-      searchInfo.innerHTML = `Displaying ${searchResult.length}/${data.length} Shows`;
-      loadEpisodes();
-  });
-
-}
-
-// Populating the episodes' page
-function episodesPage(obj) {
-
-  obj.forEach(element => {
-    
-    //Creating and appending elements
-    const expandingList = document.createElement('div', { is : 'expanding-list' });
-    expandingList.setAttribute('id', 'episodesDiv');
-    expandingList.setAttribute('class', 'expanding-div');
-    rootEpisodes.appendChild(expandingList);
-    header = document.createElement('h2');
-    expandingList.appendChild(header);
-    header.innerHTML = `${element.name} S0${element.season}E0${element.number}`;
-    header.setAttribute('class', 'name')
-    img = document.createElement('img');
-    img.setAttribute('class', 'episodeImage');
-  
-    if(element.image) {
-      img.src = element.image.medium;
-    } else {
-      img.src = "https://upload.wikimedia.org/wikipedia/commons/2/26/512pxIcon-sunset_photo_not_found.png";
-    }
-    expandingList.appendChild(img);
-
-    // Truncated summary Text
-    paragraph = document.createElement('p');
-    paragraph.setAttribute('class', 'summary');
-    expandingList.appendChild(paragraph);
-    let truncatedText;
-    if(element.summary) { 
-    truncatedText = element.summary.toString().split(' ').slice(0, 25).join(' ');
-    //Unused code that might be needed at some point
-    const truncatedText2 = element.summary.toString().split(' ').splice(25).join(' ');
-    }
-    paragraph.innerHTML = `${truncatedText} ...`;
-    let span = document.createElement('span');
-    span.setAttribute('class', 'more-text summary');
-    span.innerHTML =`${element.summary}`;
-
-    //A read more button
```

Comment:
You already created these buttons for shows. Any way of putting the code into separate functions so you can reuse it?